engines/dradis-api/lib/dradis/ce/api/warden_strategy.rb

```diff
@@ -16,12 +16,18 @@ def valid?
     end
 
     def authenticate!
-      if auth.provided? && auth.basic? && auth.credentials
-        username = auth.credentials.first
+      if token
+        user = User.enabled.find_by_api_token(token)
+        if user && secure_compare(user.api_token)
+          success!(user)
+        else
+          custom!(unauthorized(403))
+        end
+      elsif auth.provided? && auth.basic? && auth.credentials
```

Comment:
Another way we can do this is with two different warden strategies. One that accepts basic auth, and one that accepts token auth. Then at some point maybe in the controller, or in a routing strategy we can switch to the strategy we want to use. It keeps each strategy class specifics it it's _strategy_ where strategy is tokenized, or basic auth, or username/pw etc.

Up to you if you see value in that, or would prefer to keep it in a single location.

Reply:
This is now split into two strategies: `token_auth` and `basic_auth`.